src/tpm_nvfile.c

```diff
@@ -262,6 +264,8 @@ TPM_RESULT TPM_NVRAM_LoadData(unsigned char **data,     /* freed by caller */
         if (irc != 0) {
             printf("TPM_NVRAM_LoadData: Error (fatal) closing file %s\n", filename);
             rc = TPM_FAIL;
+            free(*data);
+            *data = NULL;
```

Comment:
Can you move this to the bottom:

```
if (rc && data != NULL) {
    free(*data);
    *data = NULL;
}
return rc;
```

Reply:
Updated. I skipped the `data != NULL` check since `data` is dereferenced at the beginning of the function which implies it is non-NULL.